test/integration/spec/connect.js

```diff
@@ -203,6 +204,119 @@ describe('connect', function() {
     });
   });
 
+  describe.only('media region', () => {
+    let sid = null;
+    let token = null;
+    let roomName = null;
+
+    beforeEach(async () => {
+      const identity = randomName();
+      token = getToken(identity);
+      sid = await createRoom(randomName(), defaults.topology, { MediaRegion: 'gll' });
+    });
+
+    const invalidRegions = ['foo', '34', '$abc', 'abc!'];
+    const validRegions = defaults.regions ? defaults.regions.split(',') : ['au1', 'br1', 'de1', 'ie1', 'in1', 'jp1', 'sg1', 'us1', 'us2'];
+    const mediaRegions = ['without', ...invalidRegions, ...validRegions];
+
+    mediaRegions.forEach(mediaRegion => {
+      const isInvalidRegion = invalidRegions.includes(mediaRegion);
+      const mediaRegionOptions = mediaRegion === 'without' ? {} : { mediaRegion };
+
+      let scenario = 'when called ';
+      if (isInvalidRegion) {
+        scenario += `with an invalid mediaRegion : ${mediaRegion}`;
+      } else if (mediaRegion === 'without') {
+        scenario += 'without a mediaRegion ';
+      } else {
+        scenario += `with a valid mediaRegion: ${mediaRegion}`;
+      }
+
+      context(scenario, () => {
+        it(`should return a CancelablePromise that ${isInvalidRegion ? 'rejects with a MediaServerRemoteDescFailedError if the room is not p2p' : 'resolves with a Room'}`, async () => {
+          const cancelablePromise = connect(token, Object.assign({ name: sid }, mediaRegionOptions, { tracks: [] }));
+          assert(cancelablePromise instanceof CancelablePromise);
+
+          let error = null;
+          let room = null;
+          try {
+            room = await cancelablePromise;
+          } catch (error_) {
+            error = error_;
+          } finally {
+            if (room) {
+              room.disconnect();
+              await completeRoom(sid);
+            }
+            if (isInvalidRegion) {
+              if (defaults.topology !== 'peer-to-peer') {
+                assert(error instanceof MediaServerRemoteDescFailedError, `Connected to Room ${room && room.sid} with an invalid mediaRegion "${mediaRegion}"`);
+              } else {
+                assert.equal(room.mediaRegion, null);
+              }
```

Comment:
Returning MediaServerRemoteDescFailedError when the region is invalid feels like the wrong behavior, but I understand that this is a backend and not an SDK limitation. Ideally, the Server would return [MediaRegionInvalidError](https://code.hq.twilio.com/client/twilio-video-errors/blob/master/twilio-video-errors.yaml#L200) for this error case. Perhaps its worth revisiting this now that it we are adding this new control plane to the SDK?

Reply:
I agree. I had originally implemented this test with MediaRegionInvalidError but when I got an actual error back, it was not the case. There is a thread where I asked Seth/Yuri this question and I've filed a ticket to have the documentation adjusted. 

Alternatively, I believe that this error could could also be implemented so that we wouldn't need to make changes to the documentation. I'm just not sure how to best go about this since this test seems biased at this point?